promql/ast.go

```diff
@@ -238,56 +238,58 @@ type VectorMatching struct {
 }
 
 // Visitor allows visiting a Node and its child nodes. The Visit method is
-// invoked for each node encountered by Walk. If the result visitor w is not
-// nil, Walk visits each of the children of node with the visitor w, followed
-// by a call of w.Visit(nil).
+// invoked for each node and path leading to it encountered by Walk.
```

Comment:
The path is passed as an additional parameter, it's the nodes that are getting walked - not the paths.